opentelemetry-exporters-newrelic-auto/src/main/java/com/newrelic/telemetry/opentelemetry/export/auto/NewRelicSpanExporterFactory.java

```diff
@@ -0,0 +1,38 @@
+package com.newrelic.telemetry.opentelemetry.export.auto;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.newrelic.telemetry.Attributes;
+import com.newrelic.telemetry.opentelemetry.export.NewRelicSpanExporter;
+import io.opentelemetry.sdk.contrib.auto.config.Config;
+import io.opentelemetry.sdk.contrib.auto.config.SpanExporterFactory;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+
+public class NewRelicSpanExporterFactory implements SpanExporterFactory {
+
+  private static final String INSIGHTS_INSERT_KEY = "INSIGHTS_INSERT_KEY";
+  private static final String NEWRELIC_SERVICE_NAME = "newrelic.service.name";
+  private static final String DEFAULT_NEWRELIC_SERVICE_NAME = "(unknown service)";
+
+  /**
+   * Creates an instance of a {@link SpanExporter} based on the provided configuration.
+   *
+   * @param config The configuration
+   * @return An implementation of a {@link SpanExporter}
+   */
+  @Override
+  public SpanExporter fromConfig(Config config) {
+    final String apiKey = getApiKey();
+    final String serviceName =
+        config.getString(NEWRELIC_SERVICE_NAME, DEFAULT_NEWRELIC_SERVICE_NAME);
+
+    return NewRelicSpanExporter.newBuilder()
+        .commonAttributes(new Attributes().put("service.name", serviceName))
+        .apiKey(apiKey)
+        .build();
+  }
+
+  @VisibleForTesting
+  String getApiKey() {
+    return System.getenv(INSIGHTS_INSERT_KEY);
```

Comment:
I think that the `Config` instance that is passed in here is an instance of this: https://github.com/open-telemetry/opentelemetry-auto-instr-java/blob/1f0cd54fd745fd5c914284c9a3f583faff65dddc/agent-tooling/src/main/java/io/opentelemetry/auto/tooling/DefaultExporterConfig.java#L20, so it will handle system properties/env vars/etc. so, I think this means we should always ask the Config for the value, rather than explicitly going to the environment. And, I think this means that the properties we're looking for lower-cased, dot-separated names. I'm reaching out to the auto-instrumentation group to confirm this, though.

Reply:
I was in a dilemma b/w env variable and config for `INSIGHTS_INSERT_KEY`. I updated it to read from Config. Now, It can either be passed from system property `ota.exporter.newrelic.api.key` or environment variable `OTA_EXPORTER_NEWRELIC_API_KEY`. Updated the README with the new instruction.

Let me know if the system property name needs to be changed

Reply:
let's keep it for now. Thanks!